src/NuGet.Core/NuGet.ProjectModel/PackageSpecUtility.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using NuGet.Versioning;
+
+namespace NuGet.ProjectModel
+{
+    public static class PackageSpecUtility
+    {
+        /// <summary>
+        /// Apply a snapshot value.
+        /// </summary>
+        public static NuGetVersion SpecifySnapshot(string version, string snapshotValue)
+        {
+            // Snapshots should be in the form 1.0.0-*, 1.0.0-beta-*, or 1.0.0-rc.*
+            // Snapshots may not contain metadata such as 1.0.0+5.* or be stable versions such as 1.0.*
+            if (IsSnapshotVersion(version))
+            {
+                if (string.IsNullOrEmpty(snapshotValue))
+                {
+                    version = version.Substring(0, version.Length - 2);
+                }
+                else
+                {
+                    version = version.Substring(0, version.Length - 1) + snapshotValue;
+                }
+            }
+
+            return NuGetVersion.Parse(version);
+        }
+
+        /// <summary>
+        /// True if the string is a snapshot version.
+        /// </summary>
+        public static bool IsSnapshotVersion(string version)
+        {
+            if (version != null
+                && version.EndsWith("*", StringComparison.Ordinal)
+                && version.IndexOf("-", StringComparison.Ordinal) > -1
+                && version.IndexOf("+", StringComparison.Ordinal) < 0
```

Comment:
`1.0.0-*+foo` is fine, right? We just want to be sure there is a `-*` before the plus.